### DISCUSSION TITLE
Retornando um campo Blob na API

### DISCUSSION THREAD

**Discussion by leandrosfreire** [HashLoad/horse#198] [Q&A] [answered]
Recebo um JSON de uma aplicacao que me retorna um conteudo HTML
como abaixo
{
    "products": [
        {
            "id": 4583016857673,
            "title": "BERMUDA FITNESS",
            "body_html": "<div style=\"text-align: center;\" class=\"detailmodule_text-image\" data-mce-style=\"text-align:  ...... ETC aqui segue o resto do html

eu armazeno ele no meu banco postgre como um bytea(Blob) no momento em que preciso disponibilizar ele para minha aplicacao e gero novamente e libero estou usando o datasetserialize

o problema e que quando eu disponibilizo ele vem sem os sinas de > ou = ou qualquer formatacao do html

assim

 {
        "idproduto": "6564111220809",
        "descricao": "CASE COLOR RING",
        "tags": "",
        "bodyhtml": "h3CASECOLORRING\/h3pdirltridtwtargettextdataplaceholderTraduoclasstwdatatexttwtextlargeXcVN5dtwtaspanla (aqui deveria estar formatado assim como eu recebo)

lembrando que para salvar o original no meu banco estou usando restrequest4delphi e datasetserialize tb

minha duvida e em que parte o processo esta errado, ao salvar no banco ao ler novamente enfim nao sei onde achar

**Comment by dliocode** [comment]
Olá Leandro isso é feito pelo datasetserialize, se eu não me engano, quando o campo é um blob ele converte em Base64..

Na sua aplicação cliente você vai receber o bodyhtml e vai fazer o decode64..

Veja se funciona, qualquer coisa avisa!

**Comment by viniciussanchez** [comment, marked as answer]
Como o @dliocode falou.... quem está fazendo isso é o DataSet-Serialize. Existe uma rotina nele, que quando o tipo do field é Blob, ele converte o conteúdo dele para base64 na hora de gerar o JSON. Talvez possamos criar uma configuração no DataSet-Serialize, para informar se converte ou não para Base64, porém hoje isso é padrão dele.

**Reply by leandrosfreire** [reply]
Maravilha vou testar fazendo o Decode entao

**Reply by leandrosfreire** [reply]
Encontrei 2 problemas se eu pego o conteudo do campo e faço o Decode nele retorna o seguinte erro: "No mapping for the Unicode character exists in the target multi-byte code page"

O segundo problema e que mesmo que eu pegue ele e faca o Decode e funcione eu uso o dataset serialize para enviar para o cliente direto, e na decodificacao ele fica sem a formatacao, @viniciussanchez a opcao de informar se converte ou nao seria interessante

**Comment by dliocode** [comment]
@leandrosfreire vamos fazer assim, abra uma nova discursão no repositório do [DataSet-Serialize](https://github.com/viniciussanchez/dataset-serialize/discussions), assim é mais fácil para gerenciar as discurssões. Já que não é uma situação voltada ao horse.

**Reply by leandrosfreire** [reply]
Ok Obrigado @dliocode
